src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Crypto.cs

```diff
@@ -164,5 +167,82 @@ internal static byte[] GetDynamicBuffer<THandle>(NegativeSizeReadMethod<THandle>
 
             return bytes;
         }
+
+        [LibraryImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_GetMemoryUse")]
+        internal static partial int GetMemoryUse(ref int memoryUse, ref int allocationCount);
+
+        public static int GetOpenSslAllocatedMemory()
+        {
+            int used = 0;
+            int count = 0;
+            GetMemoryUse(ref used, ref count);
+            return used;
+        }
+
+        public static int GetOpenSslAllocationCount()
+        {
+            int used = 0;
+            int count = 0;
+            GetMemoryUse(ref used, ref count);
+            return count;
+        }
+
+#pragma warning disable CA1823
+        private static readonly bool MemoryDebug = GetMemoryDebug();
+#pragma warning restore CA1823
+
+        private static bool GetMemoryDebug()
+        {
+            string? value = Environment.GetEnvironmentVariable(Interop.OpenSsl.OpenSslDebugEnvironmentVariable);
+            if (int.TryParse(value, CultureInfo.InvariantCulture, out int enabled) && enabled == 1)
+            {
+                Interop.Crypto.GetOpenSslAllocationCount();
+                Interop.Crypto.GetOpenSslAllocatedMemory();
+                Interop.Crypto.ForEachTrackedAllocation((a, b, c, d) => { });
+                Interop.Crypto.EnableTracking();
+                Interop.Crypto.DisableTracking();
+            }
+
+            return enabled == 1;
+        }
+
+        [LibraryImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EnableMemoryTracking")]
+        internal static unsafe partial void EnableMemoryTracking(int enable);
+
+        [LibraryImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_ForEachTrackedAllocation")]
+        private static unsafe partial void ForEachTrackedAllocation(delegate* unmanaged<IntPtr, int, char*, int, IntPtr, void> callback, IntPtr ctx);
+
+
+        public static unsafe void ForEachTrackedAllocation(TrackedAllocationDelegate callback)
+        {
+            GCHandle pCallback = GCHandle.Alloc(callback);
+            try
+            {
+                ForEachTrackedAllocation(&MemoryTrackingCallback, GCHandle.ToIntPtr(pCallback));
+            }
+            finally
+            {
+                pCallback.Free();
+            }
```

Comment:
```suggestion
            ForEachTrackedAllocation(&MemoryTrackingCallback, (IntPtr)&pCallback);
```
You can just take an address of the object reference. It is not necessary to use GCHandle.